src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -487,6 +498,18 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
                         {
                             nonce = ReadNonceCookie(nonce);
                         }
+
+                        var tokenValidatedContext = await RunTokenValidatedEventAsync(authorizationResponse, tokenEndpointResponse, properties, ticket, jwt, nonce);
+                        if (tokenValidatedContext.CheckEventResult(out result))
+                        {
+                            return result;
+                        }
+                        authorizationResponse = tokenValidatedContext.ProtocolMessage;
```

Comment:
It really became noisy, IMHO.
But I guess that's the price to pay if you want to keep all these properties publicly settable.
(I've never understood why, BTW, as the types themselves are already fully mutable)